packages/js/product-editor/src/blocks/notice/edit.tsx

```diff
@@ -0,0 +1,85 @@
+/**
+ * External dependencies
+ */
+import { useBlockProps } from '@wordpress/block-editor';
+import { createElement } from '@wordpress/element';
+import type { BlockAttributes, BlockEditProps } from '@wordpress/blocks';
+import { Button } from '@wordpress/components';
+import { getNewPath, navigateTo } from '@woocommerce/navigation';
+import { Product } from '@woocommerce/data';
+import { useEntityProp } from '@wordpress/core-data';
+
+/**
+ * Internal dependencies
+ */
+import { Notice } from '../../components/notice';
+import {
+	hasAttributesUsedForVariations,
+	isSelectedTabApplicableForOptionsNotice,
+} from '../../utils';
+
+export interface NoticeBlockAttributes extends BlockAttributes {
+	buttonText: string;
+	content: string;
+	id: string;
+	isSelected?: boolean;
+	title: string;
+	type: 'error-type' | 'success' | 'warning' | 'info';
+}
+
+export function Edit( {
+	setAttributes,
+	attributes,
+	context,
+}: BlockEditProps< NoticeBlockAttributes > & {
+	context?: {
+		selectedTab?: string | null;
+	};
+} ) {
+	const blockProps = useBlockProps();
+	const {
+		buttonText,
+		content,
+		id,
+		isSelected: contextIsSelected,
+		title,
+		type = 'info',
+	} = attributes;
+	const isSelected = context?.selectedTab === id;
+	if ( isSelected !== contextIsSelected ) {
+		setAttributes( { isSelected } );
+	}
+
+	const [ productAttributes ] = useEntityProp< Product[ 'attributes' ] >(
+		'postType',
+		'product',
+		'attributes'
+	);
+
+	const isOptionsNoticeVisible =
+		hasAttributesUsedForVariations( productAttributes ) &&
+		isSelectedTabApplicableForOptionsNotice( context?.selectedTab );
```

Comment:
I think we should be able to remove the `isSelectedTabApplicableForOptionsNotice` now, given it is in it's own tab anyway by the use of the `template`.

Reply:
If you want, you could use the `isInSelectedTab` context